### ISSUE TITLE
[Physics] Improve physics and collision detections

### ISSUE THREAD

**Issue by codenamecpp** [codenamecpp/carnage3d#28]
This includes:
- [ ] Cars physical dimensions
- [x] Fix peds push cars
- [x] Gravity must take into account mass of cars
- [ ] Fix damn slopes

**Comment by neuromancer** [comment]
Can you also make fences, trees and traffic signs solid? (well, it's very annoying to have solid traffic signs, but that's the original GTA..)

**Comment by codenamecpp** [comment]
Yes solid fences will be implemented but not within the scope of this task.

Trees and traffic signs I personally hate most of all in original GTA :) So they will be added far far later if added at all

**Comment by kingadami** [comment]
So I wanted to let you know that I have been working on some simple improvements to the physics of cars.  It takes into account some of the car properties to adjust the speed an handling characteristics of the car.  My task branch is [here](https://github.com/kingadami/carnage3d/tree/2-improve-car-physics).  I also have one where I have been playing with scoring.  I am still learning the layout of the code and what not so they may not be implemented the best way yet.
